Transform from 3035 to 4326

Bachibouzouk opened the pull request:
@Ludee, @christian-rli @jh-RLI  - is there a reason why is the column geom in srid 4326 and the column rea_geom_new in srid 3035 in this [register](https://openenergy-platform.org/dataedit/view/supply/ego_dp_res_powerplant)?
The conversion from 3035 into 4326 make the query for 1000 units go from 1.5 s to 30 s ... (because oedialect doesn't handle the ST_Transform unfortunately...)

Commit:
Transform from 3035 to 4326

Commit:
Update requirements

christian-rli commented:
I'm not sure why it's in a different system, but I'm going to create a new column / table with srid 4326 on Monday. Waiting for 30s is _**not**_ convenient.

christian-rli commented:
... Also, loaded powerplants in this branch don't show on the map for me. Creating a new table might not be the most elegant solution, but one that works, so I think that that's where it's going.

nesnoj commented:
> @Ludee, @christian-rli @jh-RLI - is there a reason why is the column geom in srid 4326 and the column rea_geom_new in srid 3035 in this [register](https://openenergy-platform.org/dataedit/view/supply/ego_dp_res_powerplant)?

In open_eGo we wanted to use a equal-area projection for Europe, that's why we opted for [ETRS LAEA](https://spatialreference.org/ref/epsg/3035/) for most resulting columns such as `rea_geom_new`. The other column is the original `geom` from the data sources which we retained in order to 1) keep the original data (if you transform, you cannot say this is the original data) and 2) work with other parts of the eGo data processing which require WGS84 for historical reasons.

Bachibouzouk commented:
> In open_eGo we wanted to use a equal-area projection for Europe, that's why we opted for [ETRS LAEA](https://spatialreference.org/ref/epsg/3035/) for most resulting columns such as `rea_geom_new`

I think the new column could be named `rea_geom_4326` (it should be created by transforming `rea_geom_new` from SRID 3035 into SRID 4326. I would further suggest renaming `rea_geom_new` into `rea_geom_3035` for consistency and because the word "new" does not carry a lot of meaning nor clarity ;)

nesnoj commented:
> I think the new column could be named `rea_geom_4326` (it should be created by transforming `rea_geom_new` from SRID 3035 into SRID 4326. I would further suggest renaming `rea_geom_new` into `rea_geom_3035` for consistency and because the word "new" does not carry a lot of meaning nor clarity ;)

Right.

But what do you mean by renaming? You/we cannot rename it just like that. The eGo tables must not be altered as other tools depend on it (regardless whether the name is baldy chosen or not ;).

You could copy it to schema `model_draft`, using only the recent v0.4.5 and right scenario of course. By the way, you do filter all ego data for version and scenario, right? (otherwise you'd get 1.5M * >10 versions datasets ;)

Bachibouzouk commented:
> But what do you mean by renaming

I mean rename it :) (I also said it was a suggestion). If the tools which depend on it are only within the WAM, I would be in favor to make the modification. This shouldn't be a big trouble if the column name is stored in a variable if it is used in more than one place. If the name of the database column is used in many different places in the app, this could be an actual good opportunity to replace it by a variable and make the code most robust and independent on the name of the database column. If it is used by external applications, then I would suggest we run suggestion of names by more than one people in the future :)

nesnoj commented:
> I also said it was a suggestion

Sorry, that wasn't meant to be offensive.

> If the tools which depend on it are only within the WAM..

I think only FRED and eGo_VIS uses this table but other eGo tools do too, at least DINGO, eDisGo and I guess multiple docs. Furthermore, it was created in the eGo [data_processing](https://github.com/openego/data_processing/) which is a very time-consuming (1-2 weeks) process incl. versioning, package dependencies and so on with many people involved. So renaming is actually quite hard ;). A simple copy would be much easier but that's might not what you want?!

> If the name of the database column is used in many different places in the app, this could be an actual good opportunity to replace it by a variable and make the code most robust and independent on the name of the data base column

Yes, you might want to make use of the SQLA renaming feature of columns, e.g. `processed_geom = Column('rea_geom_new', ...)` so you can use `processed_geom` in FRED instead of the db column name `rea_geom_new` ([docs ref](https://docs.sqlalchemy.org/en/13/orm/mapping_columns.html)).

By the way, `rea_geom_new` is not a simple transform but the result of a spatial reallocation 
of all renewable plants implemented by @Ludee. So this is definitely the column you want to use ;).

> The conversion from 3035 into 4326 make the query for 1000 units go from 1.5 s to 30 s

For the original problem of this issue I see 2 realistic solutions:
1. Poor: Copy the table to `model_draft` and pre-transform to 4326
2. Better: Create a [materialized view](https://www.postgresql.org/docs/10/sql-creatematerializedview.html) in the grid schema of the OEP which holds the dataset (v0.4.5 and status quo scenario) and a transformed version of the column. But this measure should be coordinated with @Ludee. If this is very urgent I could give you a hand.
Also, this solution solve the naming problem as you could define a new name in the mview.

Bachibouzouk commented:
> Sorry, that wasn't meant to be offensive.

I didn't interpreted it as offensive :)





> For the original problem of this issue I see 2 realistic solutions:

I think solution one could be implemented to have the app working fast, then we can implement solution 2 :)

christian-rli commented:
> I think the new column could be named `rea_geom_4326` (it should be created by transforming `rea_geom_new` from SRID 3035 into SRID 4326. I would further suggest renaming `rea_geom_new` into `rea_geom_3035` for consistency and because the word "new" does not carry a lot of meaning nor clarity ;)

I created a new table for now. `model_draft.openfred_powerplants`. It has two geometry columns. One is the original `geom`, the other is `rea_geom_4326`. It's the transformed former `rea_geom_new`. Hope it works with this one :)

Commit:
Use a new table for the powerplants

Commit:
Use the new table with reallocated geometries

There is no need to transform from srid 3035 to srid 4326
anymore. The column rea_geom_4326 is directly in srid 4326

Bachibouzouk committed:
Merge branch 'develop' into fix/powerplant-geometry

christian-rli reviewed (approved):
works \o/
